src/MICore/MIResults.cs

```diff
@@ -945,29 +951,51 @@ private ConstValue ParseCString(Span input, out Span rest)
             return new ConstValue(output.ToString());
         }
 
-        // return the character encoded in octal format, update the string position
-        private char DecodeOctalLiteral(string str, int i, ref int p)
+        /// <summary>
+        /// convert a string of octal encode bytes into chars using an utf8 decoder and write resulting chars to output
+        /// </summary>
+        /// <param name="str"></param>
+        /// <param name="i"></param>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        bool SpanOctalChars(string str, ref int i, StringBuilder output)
         {
-            if (i+2 >= str.Length)
+            int s = i;
+            bool error = false;
+            int cChars = 0;
+            byte[] bytes = new byte[str.Length];
+            while (!error && i + 3 < str.Length && str[i] == '\\' && str[i+1] >= '0' && str[i+1] <= '3')
             {
-                return _resultString[i];
-            }
-            int v = 0;
-            p = i;
-            for (; p < i+3; ++p)
-            {
-                char c = str[p];
-                if (c >= '0' && c <= '7')
+                int v = 0;
+                for (int n = 1; n <= 3; ++n)
                 {
-                    v = (v << 3) + (c - '0');
-                }
-                else
-                {
-                    // error, just return the character found at i
-                    return _resultString[i];
+                    char c = str[i+n];
+                    if (c >= '0' && c <= '7')
+                    {
+                        v = (v << 3) + (c - '0');
+                    }
+                    else
+                    {
+                        error = true;
+                        continue;
+                    }
                 }
+                bytes[cChars++] = (byte)v;
+                i += 4;
+            }
+            if (error)
```

Comment:
I think this check should be `if (error || cChars == 0)`. As if the input is something like `\01` the success case will wind up decrementing from `i`, which we don't want.